### PR TITLE
fix(clustering): restore json websocket protocol (#9921)

### PR DESCRIPTION
**Pull request by chronolaw**
### Summary

Try to restore WebSocket protocol and keep wRPC protocol.

Revert PR #9740, update to PR #9823  #9769 , #9616 #9671 #9774.

This is the cherry-pick of #9921, and didn't remove wRPC protocol.

See KAG-371.

Related PRs: #8926 #8357

### Checklist

- [x] The Pull Request has tests
- [x] There's an entry in the CHANGELOG

### Full changelog

* revert PR #9740
* update `control_plane.lua` with new `compat.*`
* change `compat.update_compatible_payload` to fit json+gzip
* add config item `wrpc_hybrid_protocol ` only for test

[KAG-371]: https://konghq.atlassian.net/browse/KAG-371?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: kong/clustering/init.lua [hunk 1]
```diff
@@ -44,7 +46,16 @@ function _M.new(conf)
 end
 
 
+function _M:handle_cp_websocket()
+  return self.json_handler:handle_cp_websocket()
+end
+
+
 function _M:handle_wrpc_websocket()
+  if ngx.req.get_method() == "HEAD" then -- force fallback
```
Comment (chronolaw) [inline kong/clustering/init.lua:55]: Here we force 3.0.x fallback to json + websocket.

File: kong/conf_loader/init.lua [hunk 2]
```diff
@@ -535,6 +535,7 @@ local CONF_INFERENCES = {
   untrusted_lua_sandbox_environment = { typ = "array" },
 
   legacy_worker_events = { typ = "boolean" },
+  wrpc_hybrid_protocol = { typ = "boolean" },
```
Comment (chronolaw) [inline kong/conf_loader/init.lua:538]: This flag is only for testing, should never be set to `on` in the field.

File: kong/clustering/init.lua (hunk 1 above)
Comment (chronolaw) [reply kong/clustering/init.lua]: Remove this code to support 3.10 (wRPC only).

Comment (flrgh) [reply kong/clustering/init.lua:55]: seems fine to me

**Review by flrgh — APPROVED** [review]

**Pull request merged**